ci/release/Jenkinsfile

```diff
@@ -62,13 +62,19 @@ pipeline {
 					def developmentVersion = Version.parseDevelopmentVersion(params.DEVELOPMENT_VERSION)
 					echo "Performing full release for version ${releaseVersion.toString()}"
 
-					withMaven(mavenSettingsConfig: params.RELEASE_DRY_RUN ? null : 'ci-hibernate.deploy.settings.maven',
-							mavenLocalRepo: env.WORKSPACE_TMP + '/.m2repository') {
+					withMaven(mavenSettingsConfig: null, mavenLocalRepo: env.WORKSPACE_TMP + '/.m2repository') {
```

Comment:
The goal of the custom Maven settings here was also to override the default, which uses our Nexus mirror. We probably don't want to use the Nexus mirror during releases, and use "actual" dependencies downloaded from Maven central instead. Just in case our cache was poisoned somehow (mistakes happen).

See also how we use the custom Maven settings in snapshot publishing.

Note that we indeed don't need the credentials in Maven settings anymore. We should probably remove them from these maven settings on Jenkins once we switched to JReleaser everywhere.